src/chaat/db/user.clj

```diff
@@ -0,0 +1,39 @@
+(ns chaat.db.user
+  (:require
+   [next.jdbc :as jdbc]
+   [next.jdbc.sql :as sql]))
+
+;; these 2 functions do the same thing, will remove one of them
+(defn new-user?
+  "Check if username exists in username column of user table."
+  [connection username]
+  (empty? (sql/find-by-keys connection :users {:username username})))
+
+(defn user-exists?
+  [connection username]
+  (let [result (sql/find-by-keys connection :users {:username username})]
+    (if (seq result)
+      true false)))
+
+(defn insert
+  "Insert new user into user table"
+  [db user-info]
+  (try
+    (jdbc/with-transaction [tx (db)]
+      (if (new-user? tx (:username user-info))
+        (let [query-result (sql/insert! tx :users user-info)]
+          {:result query-result :error nil})
+        {:result nil :error "Username already exists"}))
+    (catch Exception e
+      {:result nil :error (str "Postgres exception: " e)})))
+
+;; future: add authentication/authorization for deleting
+;; users need to be authenticated as username to gain authorization to delete username
```

Comment:
[nit] These comments belong to a story card/issue rather than in the code.

Reply:
Agreed. These were more a note towards reviewers who may find it weird that anyone can delete anyone. 
You mentioned earlier on about doing a self-review to point out things like this, so that would also be a good place to put notes like this.